tests/common/test_logcollector.py

```diff
@@ -0,0 +1,435 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+from datetime import datetime, timedelta
+import os
+import shutil
+import zipfile
+import tempfile
+
+from azurelinuxagent.common.utils.fileutil import rm_dirs, mkdir, rm_files, write_file
+from azurelinuxagent.common.logcollector import LogCollector
+from tests.tools import AgentTestCase, is_python_version_26, patch, skip_if_predicate_true
+
+
+SMALL_FILE_SIZE = 1 * 1024 * 1024  # 1 MB
+LARGE_FILE_SIZE = 5 * 1024 * 1024  # 5 MB
+
+
+@skip_if_predicate_true(is_python_version_26, "Disabled on Python 2.6")
+class TestLogCollector(AgentTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        AgentTestCase.setUpClass()
+
+        prefix = "{0}_".format(cls.__class__.__name__)
```

Comment:
what's the purpose of the prefix?

Reply:
Just to help with debugging, I find it easier to match the temp folder to the tests I'm interested in (when using breakpoints) since it's named after the class. We use it in a couple of other test classes too.